compiler/optimizer/LoopCanonicalizer.cpp

```diff
@@ -2087,7 +2119,41 @@ void TR_LoopCanonicalizer::eliminateRedundantInductionVariablesFromLoop(TR_Regio
          }
       }
    }
+bool TR_LoopCanonicalizer::isTrivialArrayOperation(TR::NodeChecklist &visited,const TR::Node *node)
+   {
+      if (visited.contains(node))
+         return true;
 
+      for (auto childIndex = 0; childIndex < node->getNumChildren(); ++childIndex)
+         {
+         if (isTrivialArrayOperation(visited, node->getChild(childIndex)))
+            visited.add(node);
+         else
+            {
+            traceMsg(comp(),"LoopCanonicalizer.cpp::isTrivialArrayOperation Returning from for loop %s\n", node->getChild(childIndex)->getOpCode().getName());
+            return false;         
+            }
+         }
+
+      TR::ILOpCodes opCode = node->getOpCodeValue();
+      traceMsg(comp(),"LoopCanonicalizer.cpp::isTrivialArrayOperation opCode value %d\n",opCode);
+      if (TR::ILOpCodes::bstorei = opCode
```

Comment:
This looks like a logic typo. Did you mean to use `==` here instead of `=`? I'm surprised this even compiles.

Reply:
Ah dang, that was definitely not me, it was vs code auto fill!